core/pta/attestation/attestation.c

```diff
@@ -0,0 +1,179 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) 2021, Huawei Technologies Co., Ltd
+ */
+
+#include <crypto/crypto.h>
+#include <kernel/pseudo_ta.h>
+#include <kernel/ts_store.h>
+#include <mm/file.h>
+#include <pta_attestation.h>
+#include <stdlib.h>
+#include <string.h>
+#include <tee/entry_std.h>
+#include <tee/uuid.h>
+#include <utee_defines.h>
+
+#define PTA_NAME "attestation.pta"
+
+/* Signing key is defined in signing_key.c */
+extern uint32_t pta_attestation_key_exponent;
+extern uint8_t pta_attestation_key_modulus[];
+extern size_t pta_attestation_key_modulus_size;
+extern uint8_t pta_attestation_key_private_exponent[];
+extern size_t pta_attestation_key_private_exponent_size;
```

Comment:
it it not recommended to store a private key in optee core. It mandates the optee image itslef cannot be read in clear from a boot media accessible from REE which cannot be guaranteed by OP-TEE core.

It would be better that the private key gets provisioned in core secure storage.

Reply:
Yeah for private key to be bundled in optee core, it is required to employ firmware encryption techniques like the one provided by TF-A [1].

So how about OP-TEE to rather generate this attestation key if its not already present in core secure storage during device provisioning?

[1] https://trustedfirmware-a.readthedocs.io/en/latest/design/trusted-board-boot.html#authenticated-encryption-framework

Reply:
@etienne-lms @b49020 agreed. I will update the PTA to generate the key on first call and save it to secure storage. I will also add a command to retrieve the public key.